.github/workflows/gpu-ci.yml

```diff
@@ -0,0 +1,69 @@
+name: Array API
+on:
+  workflow_dispatch:
+    inputs:
+      pr_id:
+
+  # XXX Temporary addition to execute workflow on pull requests, but only those
+  # that modify this file
+  # Remove before merging
+  pull_request:
+    paths:
+      - ".github/workflows/gpu-ci.yml"
+
+jobs:
+  tests:
+    runs-on: GPU
+    name: Run Array API unit tests
+    steps:
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
+      # change this to checkout a particular PR if a PR number is provided as input
+      - name: Checkout main repository
+        uses: actions/checkout@v2
+      - name: Checkout a particular Pull Request
+        if: inputs.pr_id
+        env:
+          PR_ID: ${{ inputs.pr_id }}
+        run: |
+          echo "Checking out" ${{ env.PR_ID }}
+          gh pr checkout ${{ env.PR_ID }}
+      - name: Cache conda environment
+        id: cache-conda
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-conda-env
+        with:
+          path: ~/conda
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('build_tools/github/pylatest_conda_forge_mkl_array-api_linux-64_conda.lock') }}
+      - name: Install miniforge
+        if: ${{ steps.cache-conda.outputs.cache-hit != 'true' }}
+        run: |
+          curl -L -O "https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-$(uname)-$(uname -m).sh"
+          bash Miniforge3-$(uname)-$(uname -m).sh -b -p "${HOME}/conda"
+          source "${HOME}/conda/etc/profile.d/conda.sh"
+          rm Miniforge3-$(uname)-$(uname -m).sh
+      - name: Install dependencies
+        if: ${{ steps.cache-conda.outputs.cache-hit != 'true' }}
+        run: |
+          source "${HOME}/conda/etc/profile.d/conda.sh"
+          # defines the get_dep and show_installed_libraries functions
+          source build_tools/shared.sh
+          conda activate base
+          conda install -c conda-forge "$(get_dep conda-lock min)" -y
+          conda-lock install --name sklearn build_tools/github/pylatest_conda_forge_mkl_array-api_linux-64_conda.lock
+      - name: Install scikit-learn
+        env:
+          # Set parallelism to 5 to overlap IO bound tasks with CPU bound tasks on GPU CI
+          # workers with 4 cores when building the compiled extensions of scikit-learn.
+          SKLEARN_BUILD_PARALLEL: 5
```

Comment:
This environment variable is not used anymore with Meson

Reply:
Should I also remove it from the Azure workflow (where I found it)?

Reply:
This can be in a separate clean-up PR I think. Note that `SKLEARN_BUILD_PARALLEL` is still used in the one `setuptools` build we still have.